src/js/components/Calendar/__tests__/Calendar-test.tsx

```diff
@@ -760,6 +760,33 @@ describe('Calendar Keyboard events', () => {
     // Jan 16th is set to active
     expect(onSelect).toBeCalledWith(expect.stringMatching(/^2020-01-16T/));
   });
+  test('onSpace', () => {
+    const onSelect = jest.fn();
+    const { getByText } = render(
+      <Grommet>
+        <Calendar date="2020-01-15" onSelect={onSelect} />
+      </Grommet>
+    );
+
+    fireEvent.mouseOver(getByText('15'));
+    fireEvent.click(getByText('15'));
```

Comment:
```suggestion
```
Can we remove these lines since we just want to test the keyboard 'space' behavior?

Reply:
i tried removing it but the test fails if we just test the space event only

Reply:
@anvesh-singh thank you we can try on our end and check!

Reply:
so we can leave the following 
```
   fireEvent.mouseOver(getByText('15'));
   fireEvent.click(getByText('15'));

```

this helps us be able to put the focus within the calendar alot easier and have it other places but to actually test the `space` actions we should also add either a keydown or left or right key then press `space` to confirm that this click the correct date


something like the following

```
    fireEvent.mouseOver(getByText('15'));
    fireEvent.click(getByText('15'));
    fireEvent.keyDown(getByText('15'), {
      key: 'ArrowRight',
      keyCode: 39,
      which: 39,
    });
    // press enter to change date to active
    fireEvent.keyDown(getByText('15'), {
      key: '',
      keyCode: 32,
      which: 32,
    });
    // Jan 16th is set to active
    expect(onSelect).toBeCalledWith(expect.stringMatching(/^2020-01-16T/));
```

Reply:
yeah but i have also added the event listener to check if the default behavior is also prevented which is a scroll down on space

Reply:
Im not sure Im following? so you want to also include to see if `window.scrollTo` has been called? We can use the keydown in the test then twice then press space?

Reply:
Don't worry about testing the preventDefault behavior in this test, I would just focus on testing the 'space' behavior

Reply:
so should i just remove it ?

Reply:
Don't remove the prevent default code in Calendar but just don't worry about testing that behavior in the unit test

Reply:
so should i remove these lines ????
@jcfilben 

const preventDefaultSpy = jest.spyOn(spaceKeyDownEvent, 'preventDefault');
expect(preventDefaultSpy).toHaveBeenCalled();

Reply:
Yes